airflow/dags/unzip_and_validate_gtfs_schedule/unzip_gtfs_schedule.py

```diff
@@ -0,0 +1,221 @@
+    bucket: ClassVar[str] = SCHEDULE_UNZIPPED_BUCKET
+    table: ClassVar[str] = "unzipping_results"
+    partition_names: ClassVar[List[str]] = ["dt"]
+    dt: pendulum.Date
+    outcomes: List[GTFSScheduleFeedExtractUnzipOutcome]
+
+    @property
+    def successes(self) -> List[GTFSScheduleFeedExtractUnzipOutcome]:
+        return [outcome for outcome in self.outcomes if outcome.success]
+
+    @property
+    def failures(self) -> List[GTFSScheduleFeedExtractUnzipOutcome]:
+        return [outcome for outcome in self.outcomes if not outcome.success]
+
+    def save(self, fs):
+        self.save_content(
+            fs=fs,
+            content="\n".join(o.json() for o in self.outcomes).encode(),
+            exclude={"outcomes"},
+        )
+
+
+def summarize_zip_contents(zip: zipfile.ZipFile, at: str = ""):
+    files = []
+    directories = []
+    for entry in zip.namelist():
+        if zipfile.Path(zip, at=entry).is_file():
+            files.append(entry)
+        if zipfile.Path(zip, at=entry).is_dir():
+            directories.append(entry)
+    return files, directories
+
+
+def process_feed_files(fs, extract, feed_directory: str = ""):
+    zipfile_files = []
+    with fs.open(extract.path) as f:
+        zip = zipfile.ZipFile(f)
+        for file in zipfile.Path(zip, at=feed_directory).iterdir():
+            if feed_directory:
+                file_name = os.path.join(feed_directory, file.name)
+            else:
+                file_name = file.name
+            with zip.open(file_name) as f:
+                file_content = f.read()
+            file_extract = GTFSScheduleFeedFile(
+                ts=extract.ts,
+                base64_url=extract.base64_url,
+                zipfile_path=extract.path,
+                original_filename=file.name,
+                # TODO: do we need to make this safe?
+                filename=file.name,
+            )
+            file_extract.save_content(content=file_content, fs=fs)
+            zipfile_files.append(file_extract)
+    return zipfile_files
+
+
+def unzip_individual_feed(
+    fs, extract: GTFSScheduleFeedExtract
+) -> GTFSScheduleFeedExtractUnzipOutcome:
+    """If zip contains a directory and nothing else at top level or files at top level,
+    we return the contents of the top level or the unique directory as the feed.
+    If there's a directory *and* other files at top level within zip, or if
+    there are nested directories, say we can't parse."""
+
+    logging.info(f"Processing {extract.name}")
+    zipfile_md5_hash = ""
+    try:
+        with fs.open(extract.path) as f:
+            zipfile_md5_hash = f.info()["md5Hash"]
+            zip = zipfile.ZipFile(f)
+        files, directories = summarize_zip_contents(zip)
+        logging.info(f"Found files: {files} and directories: {directories}")
+    except Exception as e:
+        return GTFSScheduleFeedExtractUnzipOutcome(
+            success=False,
+            zipfile_extract_md5hash=zipfile_md5_hash,
```

Comment:
`zipfile_md5_hash` won't be set on a file-not-found error, which is maybe OK but it will just tank the job instead of returning the outcome

Reply:
We discussed in Slack, `zipfile_md5_hash` is initialized as empty